packages/react-environment/src/api-providers/web3/web3-mixer-withdraw.ts

```diff
@@ -43,30 +76,142 @@ export class Web3MixerWithdraw extends MixerWithdraw<WebbWeb3Provider> {
+          case RelayedWithdrawResult.PreFlight:
+          case RelayedWithdrawResult.OnFlight:
+            this.emit('stateChange', WithdrawState.SendingTransaction);
+            break;
+          case RelayedWithdrawResult.Continue:
+            break;
+          case RelayedWithdrawResult.CleanExit:
+            this.emit('stateChange', WithdrawState.Done);
+            this.emit('stateChange', WithdrawState.Ideal);
+            transactionNotificationConfig.finalize?.({
+              address: recipient,
+              data: undefined,
+              key: 'mixer-withdraw-evm',
+              path: {
+                method: 'withdraw',
+                section: 'evm-mixer',
+              },
+            });
+            break;
+          case RelayedWithdrawResult.Errored:
+            this.emit('stateChange', WithdrawState.Failed);
+            this.emit('stateChange', WithdrawState.Ideal);
+            transactionNotificationConfig.failed?.({
+              address: recipient,
+              data: message,
+              key: 'mixer-withdraw-evm',
+              path: {
+                method: 'withdraw',
+                section: 'evm-mixer',
+              },
+            });
+            break;
+        }
+      });
+      logger.trace('Sending transaction');
+      realyedWithdraw.send(tx);
+      await realyedWithdraw.await();
+    } else {
+      logger.trace('Withdrawing without relayer');
+      this.emit('stateChange', WithdrawState.GeneratingZk);
+      const evmNote = EvmNote.deserialize(note);
+      const contract = await this.inner.getContractBySize(evmNote.amount, evmNote.currency);
+      try {
+        const zkpInputWithoutMerkleProof = fromDepositIntoZKPInput(deposit, {
+          recipient,
+          relayer: recipient,
+        });
+        const txReset = await contract.withdraw(deposit, zkpInputWithoutMerkleProof);
+        transactionNotificationConfig.loading?.({
+          address: recipient,
+          data: React.createElement('p', {}, 'Withdraw In Progress'),
+          key: 'mixer-withdraw-evm',
+          path: {
+            method: 'withdraw',
+            section: 'evm-mixer',
+          },
+        });
+        this.emit('stateChange', WithdrawState.SendingTransaction);
+        await txReset.wait();
+        transactionNotificationConfig.finalize?.({
+          address: recipient,
+          data: undefined,
+          key: 'mixer-withdraw-evm',
+          path: {
+            method: 'withdraw',
+            section: 'evm-mixer',
+          },
+        });
+        this.emit('stateChange', WithdrawState.Ideal);
+      } catch (e) {
+        const reason = await this.inner.reason(e.transactionHash);
+
+        transactionNotificationConfig.failed?.({
+          address: recipient,
+          data: reason,
+          key: 'mixer-withdraw-evm',
+          path: {
+            method: 'withdraw',
+            section: 'evm-mixer',
+          },
+        });
+        await 0;
```

Comment:
What is this await for?

Reply:
await for a tick it's something related to the event loop, will see if this is no longer needed